aloha-io-avro/src/main/scala/com/eharmony/aloha/factory/avro/factory_config.scala

```diff
@@ -0,0 +1,179 @@
+package com.eharmony.aloha.factory.avro
+
+import java.io.{File, InputStream}
+
+import com.eharmony.aloha.audit.impl.avro.{AvroScoreAuditor, Score}
+import com.eharmony.aloha.factory.ModelFactory
+import com.eharmony.aloha.factory.ex.AlohaFactoryException
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import com.eharmony.aloha.semantics.compiled.CompiledSemantics
+import com.eharmony.aloha.semantics.compiled.compiler.TwitterEvalCompiler
+import com.eharmony.aloha.semantics.compiled.plugin.avro.CompiledSemanticsAvroPlugin
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.apache.commons.io.IOUtils
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import scala.util.{Failure, Success, Try}
+
+sealed abstract class FactoryConfig {
```

Comment:
Made an `abstract class` instead of a `trait` so the companion object exposes the proper static functions but that might not be necessary. @jmorra, what do you think?  The only real reason to have the companion object is to support the magnet syntax, via the implicit definitions.